### PR TITLE
Merge feat/endpoints

### PR DESCRIPTION
**Pull request by Pederytter**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
     "require-dev": {
         "phpunit/phpunit": "^7.5",
         "laravel/laravel": "^5.8",
-        "mockery/mockery": "^1.2"
+        "mockery/mockery": "^1.2",
+        "orchestra/testbench": "^3.7"
```
Comment (rasmuscnielsen) [inline composer.json:20]: Are we using any testbench-specific features?

File: tests/Unit/EndpointTest.php [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    /** @test **/
+    public function it_doesnt_apply_append_constraints_unless_appended()
+    {
+        $endpoint = Endpoint::for(User::class)
+            ->allowedIncludes([
+                'teams', // Regular relation
+                'servers' => Endpoint::for(Server::class) // Endpoint
+                ->allowedAppends([
```
Comment (rasmuscnielsen) [inline tests/Unit/EndpointTest.php:98]: Fix indentation

File: tests/TestCase.php [hunk 3]
```diff
@@ -2,40 +2,32 @@
 
 namespace Makeable\ApiEndpoints\Tests;
 
+use Orchestra\Testbench\TestCase as OrchestraTestCase;
 
-class TestCase extends \Illuminate\Foundation\Testing\TestCase
+class TestCase extends OrchestraTestCase
```
Comment (rasmuscnielsen) [inline tests/TestCase.php:7]: Is this one being used?

File: tests/UnitTestCase.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Makeable\ApiEndpoints\Tests;
+
+use Illuminate\Contracts\Console\Kernel;
+use Makeable\ApiEndpoints\Tests\Helpers\TestHelpers;
+use Spatie\QueryBuilder\QueryBuilderServiceProvider;
+
+class UnitTestCase extends \Illuminate\Foundation\Testing\TestCase
```
Comment (rasmuscnielsen) [inline tests/UnitTestCase.php:9]: Should this just be `TestCase` ?

File: tests/Unit/EndpointTest.php [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Http\Request;
+use Makeable\ApiEndpoints\Endpoint;
+use Makeable\ApiEndpoints\Tests\Stubs\Server;
+use Makeable\ApiEndpoints\Tests\Stubs\User;
+use Makeable\ApiEndpoints\Tests\UnitTestCase;
+
+class EndpointTest extends UnitTestCase
+{
+    use RefreshDatabase;
```
Comment (rasmuscnielsen) [inline tests/Unit/EndpointTest.php:14]: I could be wrong, but I believe this test case doesn't actually hit the database. It's basically just at unit test.

However I think we should have a separate TestCase that actually *does* use the Laravel http testing api to hit a test-controller and see data returned from the database. If nothing else, to serve as a documentation for how to use the package.

Comment (Pederytter) [reply tests/Unit/EndpointTest.php:14]: These tests do make use of the parent `eagerLoadRelations` which does make a query selection to the database and therefore this is needed in order to migrate the database.

**Review by Pederytter — COMMENTED** [review]
Just a short note on the use of `RefreshDatabase;`

**Pull request merged**